### PR TITLE
Justin krs/reset password

### PR DESCRIPTION
**Pull request by justinengle-krs**
Functionality to Send Password Link and Set New Password.

The password reset message is currently from the API, as it seems as though the Api result is intended for the user, happy to change this to match the design if required.

The design does not contain 2 inputs. This change was confirmed by @matthampton-stock2shop  in keybase discussion

**Zeplin**
image
image
image

**Implemenation:**

**Default state**
image

**Password reset clicked**
image

**Set new password clicked**
image

**Passwords match**
image

**Show password**
image

**Save clicked**
image

### CODE REVIEW

File: src/repos/users-repo.ts [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+import type { ApiResult } from '@/models/api-result';
+import type PasswordResetResponse from '@/models/user/password-reset-response';
+import PasswordResetResponseDataWrapper from '@/models/user/password-reset-response-data-wrapper';
+import type SystemUser from '@/models/user/system-user';
+import SystemUserDataWrapper from '@/models/user/user-data-wrapper';
+
+export default class UsersRepo {
+  basePath = 'users';
+
+  async sendPasswordReset(username: string): Promise<ApiResult<PasswordResetResponse>> {
```
Comment (TheSuperRealJP) [inline src/repos/users-repo.ts:12]: Should these functions not reside in the CustomersRepo? The password reset resides on the Customers page, thus it leads me to believe the password reset is operating on the Customer object, The only awkward part is that these endpoints are different, `user`, not `customer`.

Comment (justinengle-krs) [reply src/repos/users-repo.ts:12]: I would have also thought so, but the endpoint exists on users, so I'd rather keep it inline with the endpoint?

Comment (TheSuperRealJP) [reply src/repos/users-repo.ts:12]: Yeah I agree. No worries, I just thought I'd raise it.

**Comment by matthampton-stock2shop** [comment]
Chromatic build: https://www.chromatic.com/build?appId=63ef6cddab1dec384359e1cd&number=10

File: src/components/Customer/Details/ContactDetails.svelte [hunk 2]
```diff
@@ -1,14 +1,110 @@
...
+    customer.workingCopy.password = '';
+    customer.workingCopy.password_confirmation = '';
+    setNewPasswordNotificationMessage = null;
+    showNewPassword = true;
+  }
+
+  async function saveChanges() {
+    let override = true;
+    if (customer.isStale) {
+      override = confirm('The customer information has changed, by saving you will override these changes.'); //confirm what this message should be
```
[the remark this answers is not in the document]
Comment (matthampton-stock2shop) [reply src/components/Customer/Details/ContactDetails.svelte:35]: @justinengle-krs Would you run this past wording past the designer? (describing the scenario)

(Happy to merge in the mean time, though)

File: src/lib/api/api.ts [hunk 3]
```diff
@@ -28,7 +33,7 @@ async function handleResponse(response: Response): Promise<any> {
     const data = await response.json();
 
     if (!response.ok) {
-      const error: string = (data && data?.message) || data?.statusText || response?.status;
+      const error: string = (data && data?.errors) || data?.statusText || response?.status;
```
[the remark this answers is not in the document]
Comment (justinengle-krs) [reply src/lib/api/api.ts:36]: The "message" was actually a copy/paste from other project mistake.....
I see the s2s returns an errors object

File: src/components/Customer/Details/ContactDetails.svelte (hunk 2 above)
[the remark this answers is not in the document]
Comment (justinengle-krs) [reply src/components/Customer/Details/ContactDetails.svelte:35]: I will do, will probably only get a response next week it seems

**Review by matthampton-stock2shop — APPROVED** [review]
Happy with this

**Pull request merged**